src/main/java/eu/dissco/core/datacitepublisher/web/HandleClient.java

```diff
@@ -0,0 +1,50 @@
+package eu.dissco.core.datacitepublisher.web;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import eu.dissco.core.datacitepublisher.exceptions.DataCiteApiException;
+import eu.dissco.core.datacitepublisher.exceptions.HandleResolutionException;
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Component;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.util.retry.Retry;
+
+@RequiredArgsConstructor
+@Component
+@Slf4j
+public class HandleClient {
+
+  @Qualifier(value = "handle")
+  private final WebClient webClient;
+
+  public JsonNode resolveHandles(List<String> handles) throws HandleResolutionException {
```

Comment:
What if this list get's very big, will it still be able to handle that or should we set a maximum on this list, or even resolve them one-by-one?

Reply:
right, the responses can actually get quite large, we saw that with the specimen processor. Added some pagination here